File: workspace/pset1/greedy.c
#include<cs50.h>
#include<stdio.h>
#include<math.h>

int main(void)
{
    float n,cents;
    int c=0;
    do
    {
        printf("O hai! How much change is owed?\n");
        n = get_float();
    }
    while(n<0);
    cents =roundf(100*n);

    while(cents>=25)
    {
        c++;
        cents = cents-25;

    }
    while(cents>=10)
    {
        c++;
        cents = cents-10;
    }
    while(cents>=5)
    {
        c++;
        cents = cents-5;

    }
    while(cents>=1)
    {
        c++;
        cents = cents -1;
    }
    printf("%i\n",c);


}

File: workspace/pset6/greedy.py
import cs50

print("O hai! ")

while True:
    print("How much change is owed?")
    n= cs50.get_float()
    if n > 0:
        break

cents = round(100*n)
count=0

if cents%25 < cents:
    q = (cents-(cents%25))//25
    count+=q
    cents=(cents%25)
if cents%10 < cents:
    q = (cents-(cents%10))//10
    count+=q
    cents=(cents%10)
if cents%5 < cents:
    q = (cents-(cents%5))//5
    count+=q
    cents=(cents%5)
if cents%1 < cents:
    q = (cents-(cents%1))//1
    count+=q
    cents=(cents%1)

print(count)

